option cc_enable_arenas = true;

import "ydb/core/protos/tx_datashard.proto";

package NKikimrDataShardLoad;
option java_package = "ru.yandex.kikimr.proto";

message TEvTestLoadRequest {
    message TLoadStop {
        optional uint64 Tag = 1;
        optional bool RemoveAllTags = 2;
    }

    message TUpdateStart {
        optional uint64 RowCount = 1;
        optional uint64 TabletId = 2;
        optional uint64 TableId = 3;
        optional uint32 Inflight = 4;

        // in some cases we need full DB path with table
        optional string Path = 6;
    }

    message TReadStart {
        optional uint64 TabletId = 1;
        optional uint64 TableId = 2;

        // Specifies the format for result data in TEvReadResult
        optional NKikimrTxDataShard.EScanDataFormat ResultFormat = 3;
    }

    optional uint64 Cookie = 1;

    // normally should be used by ut only
    optional bool NotifyWhenFinished = 2;

    oneof Command {
        TLoadStop LoadStop = 20;
        TUpdateStart UpsertBulkStart = 21;
        TUpdateStart UpsertLocalMkqlStart = 22;
        TUpdateStart UpsertKqpStart = 23;
        TUpdateStart UpsertProposeStart = 24;

        TReadStart ReadIteratorStart = 25;
    }
}

message TEvTestLoadResponse {
    optional uint32 Status = 1; // EResponseStatus from ydb/core/client/base/msgbus.h
    optional string ErrorReason = 2;
    optional uint64 Cookie = 3;

    // if load acter has been started, then it is the assigned tag
    optional uint64 Tag = 4;
}
